ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthProtocolManager.java

```diff
@@ -153,6 +156,45 @@ public EthProtocolManager(
         metricsSystem);
   }
 
+  public EthProtocolManager(
+      final Blockchain blockchain,
+      final WorldStateArchive worldStateArchive,
+      final BigInteger networkId,
+      final List<PeerValidator> peerValidators,
+      final boolean fastSyncEnabled,
+      final int syncWorkers,
+      final int txWorkers,
+      final int computationWorkers,
+      final Clock clock,
+      final MetricsSystem metricsSystem,
+      final EthProtocolConfiguration ethereumWireProtocolConfiguration,
+      final List<Long> forks) {
+    this.networkId = networkId;
+    this.peerValidators = peerValidators;
+    this.scheduler = new EthScheduler(syncWorkers, txWorkers, computationWorkers, metricsSystem);
+    this.blockchain = blockchain;
+    this.fastSyncEnabled = fastSyncEnabled;
+
+    this.shutdown = new CountDownLatch(1);
+    genesisHash = blockchain.getBlockHashByNumber(0L).get();
+
+    forkIdManager = ForkIdManager.buildCollection(genesisHash, forks, blockchain);
+
+    ethPeers = new EthPeers(getSupportedProtocol(), clock, metricsSystem);
+    ethMessages = new EthMessages();
+    ethContext = new EthContext(ethPeers, ethMessages, scheduler);
+
+    this.blockBroadcaster = new BlockBroadcaster(ethContext);
+
+    // Run validators
+    for (final PeerValidator peerValidator : this.peerValidators) {
+      PeerValidatorRunner.runValidator(ethContext, peerValidator);
+    }
+
+    // Set up request handlers
```

Comment:
I see significant overlap with one of the constructors above, can we remove some repetition here?